### PR TITLE
Instagram

### PR DESCRIPTION
**Pull request by romo-odoo**

### CODE REVIEW

File: addons/website/static/src/snippets/s_instagram_page/000.js [hunk 1]
```diff
@@ -1,81 +1,54 @@
...
-     */
-    destroy() {
-        const iframeEl = this.el.querySelector(".o_instagram_container iframe");
-        if (iframeEl) {
-            iframeEl.remove();
-            window.removeEventListener("message", this.__onMessage);
-        }
-        this._super.apply(this, arguments);
-    },
+        this.renderIframe();
```
Comment (robinlej) [inline addons/website/static/src/snippets/s_instagram_page/000.js]: Maybe render the iframe in the setup, and add the listener + set the src in `start()`

File: addons/website/static/src/snippets/s_instagram_page/instagram_page.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        // has been found with a linear regression).
+        const iframeWidth = parseInt(getComputedStyle(iframeEl).width);
+        // The profile picture is smaller when width < 432px.
+        iframeEl.height = 0.659 * iframeWidth + (iframeWidth < 432 ? 156 : 203);
+
+        this.el.querySelector(".o_instagram_container").appendChild(iframeEl);
+
+        // TODO : ...observerUnactive() / ...observerActive()
+        this.registerCleanup(() => { iframeEl.remove(); });
+    }
```
Comment (bso-odoo) [inline addons/website/static/src/snippets/s_instagram_page/instagram_page.js]: Wht not use `this.insert(...)` ?

File: addons/website/static/src/snippets/s_instagram_page/instagram_page.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    static selector = ".s_instagram_page";
+
+    setup() {
+        this.renderIframe();
+    }
+
+    start() {
+        // We have to setup the message listener before setting the src, because
+        // the iframe can send a message before this JS is fully loaded.
+        this.addListener(window, "message", this.onMessage);
```
Comment (bso-odoo) [inline addons/website/static/src/snippets/s_instagram_page/instagram_page.js]: Why not `_window.t-on-message` ?

File: addons/website/static/src/snippets/s_instagram_page/instagram_page.js (hunk 2 above)
Comment (romo-odoo) [reply addons/website/static/src/snippets/s_instagram_page/instagram_page.js]: We can do it depending on how the `observerUnactive` and `observerActive` are converted during the conversion. I will put a comment for this.

File: addons/website/static/src/snippets/s_instagram_page/instagram_page.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            "t-on-message": this.onMessage,
+        },
+    }
+
+    setup() {
+        this.renderIframe();
+    }
+
+    start() {
+        const src = `https://www.instgram.com/${this.el.dataset.instagramPage}/embed`;
```
Comment (robinlej) [inline addons/website/static/src/snippets/s_instagram_page/instagram_page.js]: inst_a_gram

File: addons/website/static/src/snippets/s_instagram_page/instagram_page.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    dynamicContent = {
+        // We have to setup the message listener before setting the src, because
+        // the iframe can send a message before this JS is fully loaded.
+        _window: {
+            "t-on-message": this.onMessage,
+        },
+    }
+
+    setup() {
+        this.renderIframe();
```
Comment (robinlej) [inline addons/website/static/src/snippets/s_instagram_page/instagram_page.js]: As you wish, but `renderIframe` is not really needed, everything could be moved directly into the `setup`

**Pull request merged**